src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -34,10 +48,49 @@ class MessageNotifier(
         .stringValue(hearingEventType.description)
         .build()
 
+    if (sizeOfMessage(hearingEvent) >= maxMessageSize) {
+      publishLargeMessage(hearingEventType = hearingEventType, hearingEvent = hearingEvent)
+      return
+    }
+
     val publishResult = topic.publish(eventType = "commonplatform.case.received", event = objectMapper.writeValueAsString(hearingEvent), attributes = mapOf("messageType" to messageTypeValue, "hearingEventType" to hearingEventTypeValue), messageGroupId = MESSAGE_GROUP_ID)
+
     log.info("Published message with message Id {}", publishResult.messageId())
   }
 
+  fun publishLargeMessage(hearingEventType: HearingEventType, hearingEvent: HearingEvent) {
+    val hearingEventTypeValue =
+      MessageAttributeValue.builder()
+        .dataType("String")
+        .stringValue(hearingEventType.description)
+        .build()
+
+    // hmpps.topic library returns SNSAsyncClient which requires S3AsyncClient
+    val snsExtendedAsyncClientConfiguration: SNSExtendedAsyncClientConfiguration = SNSExtendedAsyncClientConfiguration()
```

Comment:
It would be better to do this setup once rather than on every (large) message published.
Moving it into the library would achieve this as you suggest

Reply:
I just saw that the hmpps sqs library has a rule of thumb for contributions:

```
https://github.com/ministryofjustice/hmpps-spring-boot-sqs?tab=readme-ov-file#contribution-guidelines

As a rule of thumb new features must:

Already be implemented in several HMPPS services, i.e. at least 3
Have been running stably in a production environment, i.e. for at least 3 months
Provide value to all library consumers, i.e. this isn't the place to handle obscure edge cases
```

Reply:
i see your point of view thank you. I'll ask hmpps-dev their thoughts too. The fact it uses s3 to store messages and it's using a different library makes me think it's a new feature but either way you're right that it's something we have to handle

Reply:
Maybe initialise this once at the top of the class? I am not entirely sure what by lazy does here - might not be necessary
```suggestion
val snsExtendedClient by lazy {  AmazonSNSExtendedAsyncClient(
      topic.snsClient,
      SNSExtendedAsyncClientConfiguration = SNSExtendedAsyncClientConfiguration().withPayloadSupportEnabled(amazonS3AsyncClient, bucketName)
    )

```